### ISSUE TITLE
slow .zip uncompression between s3

### ISSUE THREAD

**Issue by tooptoop4** [piskvorky/smart_open#742]
this code is writing at around 275k records per minute, any idea how to speed it up?
my file is 6GB compressed, 180GB uncompressed. so it won't fit on local disk or memory. I notice only 200MB of memory is being consumed

```python
import io
import smart_open
import urllib.request
import zipfile
import logging
import boto3
smart_open.s3.DEFAULT_MIN_PART_SIZE = 250 * 1024**2
smart_open.s3.DEFAULT_BUFFER_SIZE = 50 * 1024**2
with io.BytesIO() as b:
    b = open('s3://redacted/biz.csv.zip', mode='rb')
    b.seek(0)
    with zipfile.ZipFile(b, mode='r') as zipf:
        for subfile in zipf.namelist():
            with smart_open.smart_open('s3://redacted/biz.csv', mode='wb') as fout:
                info = zipf.open(subfile)
                printcounter = 0
                for line in info:
                    printcounter += 1
                    info = (line)
                    _ = fout.write(info)
                    if printcounter % 500000 == 0:
                        print(str(printcounter)+' rows')
```

**Comment by rustyconover** [comment]
I know why this is slow.

The ZipFile module is doing a `seek()` on every read even if the position it wants to seek to is the current position.

smart_open when it processes a `seek()` call explicitly drops the current data buffer and sends a new request to S3 even if the destination file position matches the current position.

Performance can be improved by changing smart_open's `seek()` implementation to compare the destination position versus the current position.

I'll open a PR to improve this.

**Comment by lociko** [comment]
Any update here?

**Comment by mpenkov** [comment]
The PR is stuck because some of the tests are not passing. We are awaiting feedback from the author.

**Comment by rustyconover** [comment]
Hi All,

Just busy right now but the change has been working great so far in my
extended testing.

I’ll get to this when free.

Rusty

**Comment by JohnHBrock** [comment]
I think this can be closed now that #782 is merged.
